src/lang-promql/complete/hybrid.ts

```diff
@@ -24,47 +24,43 @@ import { AutocompleteContext, Completion, CompletionResult, snippet, SnippetSpec
 import { CompleteStrategy } from './index';
 import { Subtree } from 'lezer-tree';
 import { EditorState } from '@codemirror/next/basic-setup';
-import { promQLSyntax } from 'lezer-promql';
 import { PrometheusClient } from '../client';
+import {
+  AggregateOp,
+  aggregateOpTerms,
+  BinaryExpr,
+  binOpTerms,
+  FunctionIdentifier,
+  functionIdentifierTerms,
+  GroupingLabel,
+  GroupingLabels,
+  Identifier,
+  LabelMatcher,
+  LabelMatchers,
+  LabelName,
+  matchOpTerms,
+  MetricIdentifier,
+  StringLiteral,
+  VectorSelector,
+} from 'lezer-promql';
+
+const matchOp = 9999;
+
+function generateAutocompleteNode(): { [key: number]: AutoCompleteNode } {
```

Comment:
Why is this a function that constructs the object iteratively and not just an assignment with objectl iteral like:

```typescript
const autocompleteNode: { [key: number]: AutoCompleteNode } = {
  matchOp: ...,
  BinaryExpr: ...,
  FunctionIdentifier: ...,
  AggregateOp: ...,
}
```

Reply:
because somehow when you are doing that, typescript is considering that the key is a string and not the constant imported

Reply:
Try using computed property names, because otherwise indeed TS (but also JS would do the same) thinks you are providing a literal string key:

```typescript
const autocompleteNode: { [key: number]: AutoCompleteNode } = {
  [matchOp]: { labels: matchOpTerms, type: '' },
  [BinaryExpr]: { labels: binOpTerms, type: '' },
  [FunctionIdentifier]: { labels: functionIdentifierTerms, type: 'function' },
  [AggregateOp]: { labels: aggregateOpTerms, type: 'keyword'},
};
```

Reply:
Han that's the magic trick to use. Nice thanks a lot !

Reply:
should be ok now